internal/runner/operation_status_report.go

```diff
@@ -0,0 +1,62 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/waypoint/internal/core"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+func (r *Runner) executeStatusReportOp(
+	ctx context.Context,
+	job *pb.Job,
+	project *core.Project,
+) (*pb.Job_Result, error) {
+	app, err := project.App(job.Application.Application)
+	if err != nil {
+		return nil, err
+	}
+
+	op, ok := job.Operation.(*pb.Job_StatusReport)
+	if !ok {
+		// this shouldn't happen since the call to this function is gated
+		// on the above type match.
+		panic("operation not expected type")
```

Comment:
Best to return an error rather than panic, even though it's not expected. Unexpected things happen everyday.

Reply:
Leaving a note here, during our sync I noted that our other operations all have this pattern, so we decided to leave it in and have plans in the future to fix up all of the operations to yank out this pattern.